babbage/query/cuts.py

```diff
@@ -18,14 +19,28 @@ def cut(self, ast):
             raise QueryException('Invalid cut: %r' % ast[0])
         self.results.append((ast[0], ast[1], value))
 
+    def _check_type(self, ref, value):
```

Comment:
This method needs some py2/3 fixes (e.g unicode doesn't exist in py3 etc). You might consider using `six`.